pca.go

```diff
@@ -0,0 +1,76 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the k principal component direction vectors of
+// the matrix a projected score vectors and variance values by singular value
+// decomposition of the input. The input a is a matrix where variables are
+// represented in the columns and observations are the rows in the input.
+//
+// If weights is not nil, it is used to weight the observations. The length of
```

Comment:
Weights is used to weight the observations. If weights is nil, each weight is considered to have a value of one, otherwise the length of weight must match ...